etna/search/views.py

```diff
@@ -571,6 +576,95 @@ class WebsiteSearchView(LoginRequiredMixin, BucketsMixin, BaseFilteredSearchView
     template_name = "search/catalogue_search.html"
     title_base = "Catalogue results"
 
+    def add_insights_page_for_url(self, page: Page) -> None:
+        """
+        Finds the Insights page corresponding to the sourceUrl of a record, then adds that page to result of the same record.
+        Unmatched url is bypassed but logged.
+        """
+        slugs = [
+            result["_source"]
+            .get("@template", {})
+            .get("details", {})
+            .get("sourceUrl", "")
+            .rstrip("/")
+            .split("/")
+            .pop()
+            for result in page.object_list
+        ]
+        # filter by slug for performance boost
+        insights_page_by_url = {
+            page.get_url(self.request): page
+            for page in InsightsPage.objects.live()
+            .filter(slug__in=slugs)
+            .defer("body")
+            .select_related("teaser_image")
+        }
+        page_list = []
+        for result in page.object_list:
+            url = (
+                result["_source"]
+                .get("@template", {})
+                .get("details", {})
+                .get("sourceUrl", "")
+            )
+            url = "/" + "/".join(url.split("/")[3:])
+            if source_page := insights_page_by_url.get(url, ""):
+                result["source_page"] = source_page
+            else:
+                logger.debug(
+                    f"WebsiteSearchView:scrapped/ingested url={url} not found in insights_page_by_url={insights_page_by_url}"
+                )
+            page_list.append(result)
+        page.object_list = page_list
+
+    def add_results_page_for_url(self, page: Page) -> None:
+        """
+        Finds the Results page corresponding to the sourceUrl of a record, then adds that page to result of the same record.
+        Unmatched url is bypassed but logged.
+        """
+        slugs = [
+            result["_source"]
+            .get("@template", {})
+            .get("details", {})
+            .get("sourceUrl", "")
+            .rstrip("/")
+            .split("/")
+            .pop()
+            for result in page.object_list
+        ]
+        results_page_by_url = {
+            page.get_url(self.request): page
+            for page in ResultsPage.objects.live()
+            .filter(slug__in=slugs)
+            .select_related("teaser_image")
+        }
+        page_list = []
+        for result in page.object_list:
+            url = (
+                result["_source"]
+                .get("@template", {})
+                .get("details", {})
+                .get("sourceUrl", "")
+            )
+            url = "/" + "/".join(url.split("/")[3:])
+            if source_page := results_page_by_url.get(url, ""):
+                result["source_page"] = source_page
```

Comment:
We should use `urlparse` here:
https://docs.python.org/3/library/urllib.parse.html
```suggestion
            if source_page := insights_page_by_url.get(urlparse(url).path, ""):
                result["source_page"] = source_page
```

Reply:
Done